7_react/react1/week3/my-app/src/TodoItem/TodoItem.js

```diff
@@ -0,0 +1,74 @@
+import { useState } from "react";
+import "./TodoItem.css"
+import PropTypes from 'prop-types';
+
+export default function TodoItem(props) {
+    const [editing, setEditing] = useState(false)
+    const [description, setDescription] = useState(props.description);
+
+    const date = new Date(props.deadline)
+
+    const deadLineText = `${date.getFullYear()}-${date.getMonth()+1}-${date.getDate()}`
+
+    const decorateDone = props.done ?
+        <span><s>{props.description} | {deadLineText}</s></span>
+        : <span>{props.description } | {deadLineText}</span>
+
+    const onInputChange = (e) => {
+        setDescription(e.target.value)
+    }
+
+    const handleUpdateDescription = () => {
+        if (!editing) {
+            setEditing(true)
+        } else {
+            if (!description) {
+                window.alert('Description can\'t be empty')
+            } else {
+                props.onUpdateItem(props.id, 
+                    description,
+                    props.done
+                )
+                setEditing(false)
+            }
+        }
+    }
+
+    const handleUpdateStatus = (status) => {
+        props.onUpdateItem(props.id, 
+            props.description,
+            status
+        )
+    }
+
+    return (
+        <li key={props.taskId} className="todoItem">
```

Comment:
As a convention, class names are always written in kebab case (https://developer.mozilla.org/en-US/docs/Glossary/Kebab_case).